### ISSUE TITLE
from __future__ import annotations as <identifier> not supported

### ISSUE THREAD

**Issue by analog-cbarber** [PyCQA/pylint#3798]
In pylint 2.6.0 (and earlier) under python 3.7, the following produces an incorrect warning for
the '-> C:' return value.

```python
from __future__ import annotations as __annotations__

class C:
    @classmethod
    def create(cls) -> C: # bogus warning: Undefined variable 'C'
        return cls()

```

There is no warning when the `as` clause is not there.

This clause is required in some spaces in order to avoid having the `annotations` symbol pollute the
namespace and is not supposed to affect the processing future clause.

See https://docs.python.org/3/reference/simple_stmts.html#future

**Comment by analog-cbarber** [comment]
The problem appears to be here: 
https://github.com/PyCQA/pylint/blob/5071831a5feb608fae1a4891ccec776860d3cce4/pylint/checkers/utils.py#L1266-L1275

This check is incorrect.  I think that instead it should be checking the [future_imports attribute](http://pylint.pycqa.org/projects/astroid/en/latest/api/astroid.nodes.html#astroid.nodes.Module.future_imports)
of the astroid Module object. I think this might work:

```python
def is_postponed_evaluation_enabled(node: astroid.node_classes.NodeNG) -> bool: 
     """Check if the postponed evaluation of annotations is enabled""" 
     module = node.root()
     return "annotations" in module.future_imports
```

**Comment by AWhetter** [comment]
I can reproduce this on master with Python 3.8 as well. Your proposed correction looks like the right fix to me too.
